addons/sale/models/sale.py

```diff
@@ -1635,8 +1642,15 @@ def product_uom_change(self):
                 uom=self.product_uom.id,
                 fiscal_position=self.env.context.get('fiscal_position')
             )
-            company = self.company_id or self.order_id.company_id
-            self.price_unit = self.env['account.tax']._fix_tax_included_price_company(self._get_display_price(product), product.taxes_id, self.tax_id, company)
```

Comment:
The `price_unit` is not supposed to be dynamically recomputed depending of the taxes, only with the fiscal position.

Reply:
It is because the pricelist rules applied depends on the product, qty & uom, which means the price is recomputed as well when the qty/uom is modified.